internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+										regexp.MustCompile("^[a-zA-Z0-9-]{3,15}$"),
+										"the cluster name must be 3-15 characters long and contain only letters, numbers and hyphens",
+									),
+								},
+
+								"azure_service_endpoint": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+
+								"cloud_account_name": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+
+								"witness_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										"Cloud",
+										"FileShare",
+									}, false),
+								},
+
+								"witness_path": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+							},
+						},
+					},
+
+					"domain_fqdn": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"host_network": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"intent": {
+									Type:     pluginsdk.TypeList,
+									Required: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"name": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringIsNotEmpty,
+											},
+
+											"adapter": {
+												Type:     pluginsdk.TypeList,
+												Required: true,
+												ForceNew: true,
+												MinItems: 1,
+												Elem: &pluginsdk.Schema{
+													Type:         pluginsdk.TypeString,
+													ValidateFunc: validation.StringIsNotEmpty,
+												},
+											},
+
+											"traffic_type": {
+												Type:     pluginsdk.TypeList,
+												Required: true,
+												ForceNew: true,
+												MinItems: 1,
+												Elem: &pluginsdk.Schema{
+													Type: pluginsdk.TypeString,
+													ValidateFunc: validation.StringInSlice([]string{
+														"Compute",
+														"Storage",
+														"Management",
+													}, false),
+												},
+											},
+
+											"adapter_property_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"jumbo_packet": {
```

Comment:
should this be an int? and validated?

Reply:
It's plain string format and related to specific OEM setting. https://github.com/Azure/azure-rest-api-specs/blob/c41b76298becce0a8cfac4e682773d5d0c78dcb2/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/deploymentSettings.json#L283-L285